File: CS50/sentimental-readability/readability.py
from cs50 import get_string

# index = 0.0588 * L - 0.296 * S - 15.8
# L is average number of letter per 100 words (L/W * 100), S is avg number of sentences per 100 words (S/W * 100)


def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    index = (
        0.0588 * (float(letters) / words) * 100
        - 0.296 * (float(sentences) / words) * 100
        - 15.8
    )

    if index >= 1 and index < 16:
        index = round(index)
        print(f"Grade {index}")
    elif index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")


def count_letters(text):
    letters = 0
    for letter in text:
        if letter.isalpha():
            letters += 1
    return letters


def count_words(text):
    words = 1
    for char in text:
        if char.isspace() == True:
            words += 1
    return words


def count_sentences(text):
    sentences = 0
    for char in text:
        if char == "." or char == "!" or char == "?":
            sentences += 1
    return sentences


main()

File: CS50/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// index = 0.0588 * L - 0.296 * S - 15.8
//  L is average number of letter per 100 words (L/W * 100), S is avg number of sentences per 100 words (S/W * 100)
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    double index = 0.0588 * ((float) letters / words) * 100 - 0.296 * ((float) sentences / words) * 100 - 15.8;
    if (index >= 1 && index <= 15)
    {
        index = round(index);
        printf("Grade %i\n", (int)index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    // printf("%i letters\n", letters);
    // printf("%i words\n", words);
    // printf("%i sentences\n", sentences);
}

int count_letters(string text)
{
    int text_length = strlen(text);
    int letters = 0;
    for (int i = 0; i < text_length; i++)
    {
        if (isalpha(text[i])) // Start here
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int text_length = strlen(text);
    int words = 1;
    for (int i = 0; i < text_length; i++)
    {
        if (isspace(text[i])) // Start here
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int text_length = strlen(text);
    int sentences = 0;
    for (int i = 0; i < text_length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?') // Start here
        {
            sentences++;
        }
    }
    return sentences;
}
